openpectus/engine/main.py

```diff
@@ -82,7 +90,12 @@ async def main_async(args):
         return
 
     engine = Engine(uod, enable_archiver=True)
-    dispatcher = EngineDispatcher(f"{args.aggregator_hostname}:{args.aggregator_port}", uod.options)
+
+    # select port based on --secure, but only if --aggregator_port has the default value
+    port = default_port_secure if args.secure and args.aggregator_port == default_port \
```

Comment:
It's not a big deal, but this will make it impossible to run securely on port 9800, even if setting `-s -ap 9800`.
If we removed the default from the argument parsing above, and checked `aggregator_port` against `None` here, (and defaulted to 9800 if None and not secure here), we could support all ports.